components/EditModal.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+  Dispatch,
+  FormEvent,
+  SetStateAction,
+  useEffect,
+  useState,
+} from 'react';
+import { Gift } from '~/pages';
+import { Modal } from './Modal';
+import { TitleText } from './TitleText';
+import { Button } from './Button';
+import { updateGift } from '~/utils/jsonServerFunctions';
+import { Input } from './Input';
+import { SvgCheckMarkIcon } from '~/icons/CheckMarkIcon';
+import { SvgDeclineIcon } from '~/icons/DeclineIcon';
+
+type EditModal = {
+  gift: Gift;
+  refreshGiftList: () => void;
+  setIsModalOpen: Dispatch<SetStateAction<boolean>>;
+};
+
+export function EditModal({
+  gift,
+  refreshGiftList,
+  setIsModalOpen,
+}: EditModal) {
+  const [giftReceiver, setGiftReceiver] = useState('');
+  const [giftName, setGiftName] = useState('');
+
+  useEffect(() => {
+    setGiftName(gift.gift);
+    setGiftReceiver(gift.name);
+  }, [gift]);
+
+  useEffect(() => {
+    function handleKeyDown(e: KeyboardEvent) {
+      if (e.key === 'Escape') {
+        refreshGiftList();
+        setIsModalOpen(false);
+      }
+    }
+    document.addEventListener('keydown', handleKeyDown);
+    return function clearFunctions() {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [refreshGiftList, setIsModalOpen]);
+
+  async function handleEdit(e: FormEvent<HTMLElement>) {
+    e.preventDefault();
+
+    const newGift = gift;
+    newGift.name = giftReceiver;
+    newGift.gift = giftName;
+    await updateGift(gift.id, newGift).catch(() => {
```

Comment:
Nyt sulla `updateGift` päivitää koko lahja-objektin tietokannassa `put` metodilla:
```ts
export async function updateGift(id: string, newObject: Gift) {
  await axios.put(`${baseURL}/${id}`, newObject);
}
```
Sun käyttötarkoitukseen voisi olla `patch` metodi parempi, joka päivitää vain ne kentät mitä antaa. Eli jos `updateGift` muokkaisi näin:
```ts
export async function updateGift(id: string, newObject: Partial<Gift>) {
  await axios.patch(`${baseURL}/${id}`, newObject);
}
```

Niin sitten tämän kutsun voisi täällä muuttaa
```ts
await updateGift(gift.id, { name: giftReceiver, gift: giftName })
```

Reply:
En ollut tuosta kuullutkaan. Hyvä nosto! Jotenkin ajattelin, että GET, POST, PUT ja DELETE olisivat ainoat request-tavat, vaikka onhan siellä muitakin (OPTIONS, HEAD, PATCH, yms) :smile: